test/preprocessed/MgvGlobalTest.post.sol

```diff
@@ -0,0 +1,161 @@
+
+contract MgvGlobalTest is Test {
+
+  function cast(uint u, uint8 to) internal pure returns (uint) {
+    return u & (type(uint).max >> (256-to));
+  }
+
+  function test_pack(address monitor, bool useOracle, bool notify, uint gasprice, uint gasmax, bool dead) public {
+    MgvStructs.GlobalPacked packed = MgvStructs.Global.pack(monitor, useOracle, notify, gasprice, gasmax, dead);
+    assertEq(packed.monitor(),monitor,"bad monitor");
+    assertEq(packed.useOracle(),useOracle,"bad useOracle");
+    assertEq(packed.notify(),notify,"bad notify");
+    assertEq(packed.gasprice(),cast(gasprice,16),"bad gasprice");
+    assertEq(packed.gasmax(),cast(gasmax,24),"bad gasmax");
+    assertEq(packed.dead(),dead,"bad dead");
+  }
+
+  /* test_set_x tests:
+     - setting works
+     - get(set(get(x))) = get(x)
+     - dirty bit cleaning 
+     - no additional bits being dirtied
+  */
+  function test_set_monitor(MgvStructs.GlobalPacked packed,address monitor) public {
+      MgvStructs.GlobalPacked original = packed.monitor(packed.monitor());
+      assertEq(original.monitor(),packed.monitor(), "original: bad monitor");
+
+      MgvStructs.GlobalPacked modified = packed.monitor(monitor);
+
+      assertEq(modified.monitor(),monitor,"modified: bad monitor");
+
+      assertEq(modified.useOracle(),modified.useOracle(),"modified: bad useOracle");
+      assertEq(modified.notify(),modified.notify(),"modified: bad notify");
+      assertEq(modified.gasprice(),modified.gasprice(),"modified: bad gasprice");
+      assertEq(modified.gasmax(),modified.gasmax(),"modified: bad gasmax");
+      assertEq(modified.dead(),modified.dead(),"modified: bad dead");
+    }
+  function test_set_useOracle(MgvStructs.GlobalPacked packed,bool useOracle) public {
+      MgvStructs.GlobalPacked original = packed.useOracle(packed.useOracle());
+      assertEq(original.useOracle(),packed.useOracle(), "original: bad useOracle");
+
+      MgvStructs.GlobalPacked modified = packed.useOracle(useOracle);
+
+      assertEq(modified.useOracle(),useOracle,"modified: bad useOracle");
+
+      assertEq(modified.monitor(),modified.monitor(),"modified: bad monitor");
+      assertEq(modified.notify(),modified.notify(),"modified: bad notify");
+      assertEq(modified.gasprice(),modified.gasprice(),"modified: bad gasprice");
+      assertEq(modified.gasmax(),modified.gasmax(),"modified: bad gasmax");
+      assertEq(modified.dead(),modified.dead(),"modified: bad dead");
+    }
+  function test_set_notify(MgvStructs.GlobalPacked packed,bool notify) public {
+      MgvStructs.GlobalPacked original = packed.notify(packed.notify());
+      assertEq(original.notify(),packed.notify(), "original: bad notify");
+
+      MgvStructs.GlobalPacked modified = packed.notify(notify);
+
+      assertEq(modified.notify(),notify,"modified: bad notify");
+
+      assertEq(modified.monitor(),modified.monitor(),"modified: bad monitor");
```

Comment:
Should modified be packed on one side og the assert? This seems to only assert get does not modify?

Reply:
Thanks! goal was to ensure that the setter above did not modify the other values. The `modified` on the right should be `packed`.